core/templates/components/ck-editor-helpers/ck-editor-4-rte.component.ts

```diff
@@ -44,21 +45,45 @@ interface RteConfig extends CKEDITOR.config {
             '</div></div>',
   styleUrls: []
 })
-export class CkEditor4RteComponent implements AfterViewInit, OnDestroy {
+export class CkEditor4RteComponent implements AfterViewInit, OnDestroy, OnInit {
   @Input() uiConfig: UiConfig;
   @Input() value;
   @Input() headersEnabled = false;
   @Output() valueChange: EventEmitter<string> = new EventEmitter();
   rteHelperService;
   ck: CKEDITOR.editor;
+  hasNetworkConnection: boolean;
+  hasInternetAccess: boolean;
+  status: string = 'ONLINE';
+  disableOnOfflineNames: string[] = [
+    'image', 'skillreview', 'svgdiagram', 'video'];
   constructor(
     private ckEditorCopyContentService: CkEditorCopyContentService,
     private contextService: ContextService,
-    private elementRef: ElementRef
+    private elementRef: ElementRef,
+    private connectionService: ConnectionService
   ) {
     this.rteHelperService = OppiaAngularRootComponent.rteHelperService;
   }
 
+  ngOnInit(): void {
+    this.connectionService.onInternetStateChange.subscribe(currentState => {
+      this.hasNetworkConnection = currentState.hasNetworkConnection;
+      this.hasInternetAccess = currentState.hasInternetAccess;
+      if (this.hasNetworkConnection && this.hasInternetAccess) {
+        if (this.status === 'OFFLINE') {
+          this.enableRTEicons();
+        }
+        this.status = 'ONLINE';
+      } else {
+        if (this.status === 'ONLINE') {
```

Comment:
Is this check required if the previous check is for 'OFFLINE'? Can "status" take other values too?

Reply:
No, but as we send requests continuously the emitter will emit offline event again and again periodically, therefore we need this check to ensure that we are applying changes only when we get the first offline event and not after emit.